### PR TITLE
AJ Tran

### PR DESCRIPTION
**Pull request by ajtran303**
I completed all the stories and learned a cool way to write my tests!

### CODE REVIEW

File: app/controllers/projects_controller.rb [hunk 1]
```diff
@@ -1,8 +1,5 @@
 class ProjectsController < ApplicationController
   def show
-    project = Project.find(params[:id])
-    @project_name = project.name
-    @project_material = project.material
-    @project_theme = project.challenge.theme
+    @project = Project.find(params[:id])
   end
```
Comment (ajtran303) [inline app/controllers/projects_controller.rb:4]: I think I should have titled this commit "Refactor projects_controller show action."

File: app/models/project.rb [hunk 2]
```diff
@@ -1,4 +1,12 @@
 class Project <ApplicationRecord
   validates_presence_of :name, :material
   belongs_to :challenge
+
+  has_many :contestant_projects
+  has_many :contestants, through: :contestant_projects
+
+  def average_contestant_experience
```
Comment (corywest) [inline app/models/project.rb:8]: Look into using ActiveRecord's `average` function here instead.

File: app/views/contestants/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+<% @contestants.each do |contestant| %>
+  <%= tag.h2 contestant.name %>
```
Comment (corywest) [inline app/views/contestants/index.html.erb:2]: You can wrap your erb snippets instead of using `tag.h2`

For example: 
`<h2><%= contestant.name %></h2>`

**Comment by corywest** [comment]
Looks good overall. Take a look at my comments above!

File: app/models/project.rb (hunk 2 above)
Comment (ajtran303) [reply app/models/project.rb:8]: Thanks! I also was not using .sum right, haha.

```ruby
total_experience = contestants.sum(&:years_of_experience).to_f
```

**Pull request closed without merging**